src/Eshopworld.WorkerProcess/Stores/CosmosDbLeaseStore.cs

```diff
@@ -86,6 +86,10 @@ public async Task<LeaseStoreResult> TryUpdateLeaseAsync(ILease lease)
                 {
                     return new LeaseStoreResult(null, false);
                 }
+                catch (DocumentClientException ex) when (ex.StatusCode == HttpStatusCode.PreconditionFailed)
```

Comment:
minor comment - we could merge the two when conditions into one as the difference is the status code only - handling is the same